doc/api/report.md

```diff
@@ -0,0 +1,208 @@
+# Diagnostic Report
+
+Delivers a JSON-formatted Diagnostic summary, written to a file.
+
+The report is intended for development, test and production use, to capture
+and preserve information for problem determination. It includes JavaScript
+and native stack traces, heap statistics, platform information, resource
+usage etc. With the report option enabled, diagnostic reports can be triggered
+on unhandled exceptions, fatal errors and user signals, in addition to
+triggering programatically through API calls.
+
+## Usage
+
+```bash
+node --experimental-report --diagnostic-report-uncaught-exception \
+  --diagnostic-report-on-signal --diagnostic-report-on-fatalerror app.js
+```
+
+* `--experimental-report` Enables the diagnostic report feature.
+ In the absence of this flag, use of all other related options will result in
+ an error.
+
+* `--diagnostic-report-uncaught-exception` Enables report to be generated on
```

Comment:
Why are the flags prefixed with `--diagnostic`? It seems a bit too verbose and easy to type wrong for me..

It would be nice to be able to specify the configurations through a file instead of having to do this for every key - but I think that can be left as a future exercise.

Reply:
Probably it is good to have a consensus on the user flags as well. I just modified the flag names to include `diagnostic` following @jasnell 's proposal https://github.com/nodejs/node/pull/22712#pullrequestreview-186591273